judge/models/contest.py

```diff
@@ -122,12 +128,29 @@ def format_class(self):
     def format(self):
         return self.format_class(self, self.format_config)
 
+    @cached_property
+    def get_label_for_problem(self):
+        def DENY_ALL(obj, attr_name, is_setting):
+            raise AttributeError()
+        lua = LuaRuntime(attribute_filter=DENY_ALL, register_eval=False, register_builtins=False)
+        return lua.eval(self.problem_label_script or settings.DMOJ_CONTEST_DEFAULT_PROBLEM_LABEL_SCRIPT)
```

Comment:
I think this should be queried from the contest format. All ICPC contests should have letters, for instance.